contracts/token/ERC721/ERC721.sol

```diff
@@ -209,14 +209,24 @@ contract ERC721 is ERC165, IERC721 {
     /**
      * @dev Internal function to burn a specific token
      * Reverts if the token does not exist
-     * @param tokenId uint256 ID of the token being burned by the msg.sender
+     * Deprecated, use _burn(uint256) instead.
+     * @param tokenId uint256 ID of the token being burned
      */
     function _burn(address owner, uint256 tokenId) internal {
         _clearApproval(tokenId);
         _removeTokenFrom(owner, tokenId);
         emit Transfer(owner, address(0), tokenId);
     }
 
+    /**
+     * @dev Internal function to burn a specific token
+     * Reverts if the token does not exist
+     * @param tokenId uint256 ID of the token being burned
+     */
+    function _burn(uint256 tokenId) internal {
+        _burn(ownerOf(tokenId), tokenId);
```

Comment:
If we want this to result in a performance improvement (one less `SLOAD`!) it should be the other way around: the old `_burn(owner, tokenId)` should be implemented in terms of the new one.

Reply:
Oh wait, we do need to call `_removeTokenFrom`, right?

Reply:
Yes, we do need to know the address, what we don't need is to provide it (since it will always be `ownerOf(tokenId)`. The actuall access control check happens in `ERC721Burnable`, and is

```
require(_isApprovedOrOwner(msg.sender, tokenId));
```